microcosm_postgres/dag.py

```diff
@@ -101,16 +101,39 @@ class DAGCloner(metaclass=ABCMeta):
     def __init__(self, graph):
         pass
 
-    def explain(self, **kwargs):
+    @property
+    def anonymizers(self):
+        """
+        Model-based modifying functions that remove sensitive data when cloning
+
+        """
+        return dict()
+
+    def explain(self, anonymize=False, **kwargs):
         """
         Generate a "dry run" DAG of that state that WILL be cloned.
 
         """
         root = self.retrieve_root(**kwargs)
         children = self.iter_children(root, **kwargs)
         dag = DAG.from_nodes(root, *children)
+        if anonymize:
```

Comment:
Can we make the anonymize function decide if it should be a noop? And eliminate the branch here?